### DISCUSSION TITLE
CultureDataFormatter in TypeScript

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [SuperOffice/typescript-for-superoffice#22] [General]
Hi!

What's the best practice for handling Cultures when using for example getArchiveAgent?

For example, if we get "displayValue": "[I:5]", do we need to use CultureDataFormatter.ParseEncodedInt, or is there any better way?

**Comment by PPetterssonSO** [comment]
When I started playing around, how do we in fact get hold of CultureDataFormatter.ParseEncodedInt? :)

let test = "[I:5]";
let myNumber = CultureDataFormatter.ParseEncodedInt(test);;

Returns Cannot find name 'CultureDataFormatter'.

**Comment by PPetterssonSO** [comment]
Nevermind, built my own :)

**Comment by SuperOfficeDevNet** [comment]
We will look at getting this into the library by the time we ship it.

[CultureDataFormatter.zip](https://github.com/user-attachments/files/18199200/CultureDataFormatter.zip)

Example usage...

import { CultureDataFormatter } from './CultureDataFormatter';

console.log('CultureDataFormatter encode int: ' + CultureDataFormatter.encodeInt(123));
console.log('CultureDataFormatter parse int: ' + CultureDataFormatter.parseInteger('[I:123]'));

const date = new Date();

console.log('CultureDataFormatter encode datetime: ' + CultureDataFormatter.encodeDateTime(date));
console.log('CultureDataFormatter parse datetime: ' + CultureDataFormatter.parseDateTime('[DT:2021-09-01T14:11:45.100Z]'));

const time = CultureDataFormatter.encodeTime(date);
console.log('CultureDataFormatter encode time: ' + time);
console.log('CultureDataFormatter parse time (UTC): ' + CultureDataFormatter.parseTime(time));
